src/main/scala/org/tessellation/schema/Hom.scala

```diff
@@ -177,28 +174,78 @@ trait Hom[+A, +B] extends Ω {}
 //    )
 //}
 
+/*
+ *
+ */
 //todo Cv algebras here
-abstract class Cell[M[_]: Monad, F[_]: Traverse, A, B, C](
-  data: A, // L1Edge(txs)
-  // input: G[A], A => G[A]
-  algebra: AlgebraM[M, F, B],
-  coalgebra: CoalgebraM[M, F, C]
-) extends Topos[A, B] {
-  protected def hyloM(implicit input: A => C) = scheme.hyloM(algebra, coalgebra).apply(input(data))
+
+class Cell[M[_]: Monad, F[_]: Traverse, A, B, S](val data: A, val hylo: S => M[B], val convert: A => S)
+    extends Topos[A, B] {
+  def run(): M[B] = hylo(convert(data))
 }
 
 object Cell {
-  //  def create(data, algebra, coalgebra)
-  //  implicit val arrowInstance: Arrow[Cell] = new Arrow[Cell] {
-  //    override def lift[A, B](f: A => B): Cell[A, B] = ???
-  //
-  //    override def first[A, B, C](fa: Cell[A, B]): Cell[(A, C), (B, C)] = ???
-  //
-  //    override def compose[A, B, C](f: Cell[B, C], g: Cell[A, B]): Cell[A, C] =
-  //      // A ---> B ---> C
-  //      f.run(g.run(g)).asInstanceOf[Cell[A, C]]
-  ////      Cell(g.transform).transform
-  //  }
+  case class NullTerminal() extends Ω
+
+  def unapply[M[_], F[_], A, B, S](cell: Cell[M, F, A, B, S]): Some[(A, S => M[B])] = Some(cell.data, cell.hylo)
+
+  implicit def toCell[M[_], F[_], A <: Cell[M, F, Ω, Either[CellError, Ω], Ω]](
+    a: A
+  ): Cell[M, F, Ω, Either[CellError, Ω], Ω] = a.asInstanceOf[Cell[M, F, Ω, Either[CellError, Ω], Ω]]
+
+  implicit def cellMonoid[M[_]: Applicative, F[_]: Applicative](
+    implicit M: Monad[M],
+    F: Traverse[F]
+  ): Monoid[Cell[M, F, Ω, Either[CellError, Ω], Ω]] =
+    new Monoid[Cell[M, F, Ω, Either[CellError, Ω], Ω]] {
```

Comment:
I thought that we need `MonoidK` for Cell since we're combining the structure, not the values? But if that's not the case I think we'll need a `Monoid` or at least `Semigroup` for the value type, since a regular `Monoid` should also combine values.

Reply:
The question is then which type of `Cell` needs to be `*` to implement `MonoidK`. The implementation under the hood will be the same I guess for both `MonoidK` and `Monoid`. Such situation happens for example for `Monoid[List[Int]]` and `MonoidK[List].empty[Int]`. If we use `MonoidK` them we need to fight more with casting as  `A` type passed to `empty[A]` and `combineK[A]` need to be the same for all the cells (sooo there must be a common type so we can cast both As to same type...). So in the end we will just do `empty[Ω]` and `comineK[Ω]` which won't differ from `Monoid`. We use `ΩList extends Ω` as common type for all the `Ω`.